### PR TITLE
refactor: move addLogin method to Billing module

### PR DESCRIPTION
**Pull request by mantariksh**
## Problem

The `addLogin` function is part of the SPCP module, but it really belongs to the Billing module as it relates to tracking logins for e-service IDs. This is blocking #1175 as the MyInfo module needs to share the code for adding logins.

## Solution

Move the `addLogin` function to the Billing module.

### CODE REVIEW

File: src/app/modules/billing/__tests__/billing.factory.spec.ts [hunk 1]
```diff
@@ -39,6 +45,18 @@ describe('billing.factory', () => {
         expect(actualResults._unsafeUnwrap()).toEqual([])
       })
     })
+
+    describe('addLogin', () => {
+      it('should return MissingFeatureError', async () => {
```
Comment (liangyuanruo) [inline src/app/modules/billing/__tests__/billing.factory.spec.ts:50]: nitpick - `it` statements should explain the context that they are testing e.g. "it should return MissingFeatureError if provided with an invalid form object".

that said, should this particular test be left to the type system instead?

Comment (liangyuanruo) [reply src/app/modules/billing/__tests__/billing.factory.spec.ts:50]: we could also take this opportunity to rename the `addLogin` function for better clarity e.g. `recordLoginByForm`

**Review by liangyuanruo — APPROVED** [review]
only nitpicks, LGTM otherwise. thanks!

File: src/app/modules/billing/__tests__/billing.factory.spec.ts (hunk 1 above)
Comment (mantariksh) [reply src/app/modules/billing/__tests__/billing.factory.spec.ts:50]: >nitpick - it statements should explain the context that they are testing e.g. "it should return MissingFeatureError if provided with an invalid form object".

this test actually doesn't require any qualifications, as it is testing that the function _always_ returns `MissingFeatureError`. if you look at the context of the nesting `describe` block, this test is for the case when `spcp-myinfo feature disabled`.

> that said, should this particular test be left to the type system instead?

could you explain what you mean by this? factories are always types to return either the service function (if the feature is enabled) or a `MissingFeatureError`/passthrough function (if the feature is disabled). this test is checking that the factory correctly provides a function which only returns a `MissingFeatureError` if the `spcp-myinfo` feature is disabled.

**Pull request merged**